src/duckling/selection/selection.service.ts

```diff
@@ -23,82 +28,137 @@ export interface Selection {
-        this._store.dispatch(_selectionAction(entityKey), mergeKey);
+    select(entityKeys : EntityKey[] = [], mergeKey? : any) {
+        this._store.dispatch(_selectionAction(entityKeys), mergeKey);
     }
 
     /**
      * Clear the current selection.
      * @param  mergeKey  Optional mergeKey, describes which commands the action should be merged with.
      */
     deselect(mergeKey? : any) {
-        this.select("", mergeKey);
+        this.select([], mergeKey);
     }
 
-    private _getSelection(selectedEntity : EntityKey) : Selection {
-        if (!selectedEntity) {
-            return {};
+    getEntityKeyAtPosition(position: Vector) : EntityKey {
+        let entities = Array.from(this._renderPriority.sortEntities(this._entitySystem.entitySystem.getValue()));
+        entities.reverse();
+        let taggedEntity = entities
+            .filter(entity => this._drawerService.isEntityVisible(entity.entity))
+            .find(entity => this._entityContainsPoint(entity.key, position));
+
+        if (taggedEntity) {
+            return taggedEntity.key;
+        } else {
+            return null;
         }
+    }
+
+    getEntityKeysAtArea(position : Vector, dimension : Vector) : EntityKey[] {
+        let entities = Array.from(this._renderPriority.sortEntities(this._entitySystem.entitySystem.getValue()));
+        entities.reverse();
+        let taggedEntities = entities
+            .filter(entity => this._drawerService.isEntityVisible(entity.entity))
+            .filter(entity => this._entityContainsWithinArea(entity.key, position, dimension));
 
-        let entity = this._entitySystem.getEntity(selectedEntity);
-        if (!this._drawerService.isEntityVisible(entity)) {
-            return {};
+        if (taggedEntities) {
+            return taggedEntities.map(entity => entity.key);
+        } else {
+            return null;
         }
+    }
 
-        return {
-            selectedEntity,
-            entity : entity || null
-        };
+    isSelected(entityKey : EntityKey) {
+        return this.selections.value.find(selection => selection.key === entityKey) !== undefined;
+    }
+
+    private _entityContainsPoint(entityKey : EntityKey, position : Vector) {
+        return boxContainsPoint(this._entityBoxService.getEntityBox(entityKey), position);
     }
+
+    private _entityContainsWithinArea(entityKey : EntityKey, position : Vector, dimension : Vector) {
+        return boxContainsBox(
+            normalizeBox({
+                position,
+                dimension,
+                rotation: 0
+            }),
+            this._entityBoxService.getEntityBox(entityKey));
+    }
+
+    private _getSelections(selectedEntityKeys : string[]) : Selection[] {
+        let filteredEntities: Selection[] = [];
+        if (selectedEntityKeys && selectedEntityKeys.length > 0) {
+            for (let key of selectedEntityKeys) {
+                let entity = this._entitySystem.getEntity(key);
+                if (!entity) {
```

Comment:
Hmmm. I didn't see any reason. I removed that check, as well as the if statement that was wrapping the for loop, as it just won't run an iteration if the selectedEntityKeys array is empty. I assume that we had the check in because we were previously initializing the "selections" object on the state incorrectly.
@tlein Should verify that these weren't necessary.